libff/algebra/curves/curve_serialization.tcc

```diff
@@ -0,0 +1,184 @@
+/** @file
+ *
+ * This is a modified version of code that originaly appeared in
+ * https://github.com/clearmatics/zeth.
+ *
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
```

Comment:
I changd this notice a little bit for these files which are completely new code, and added the notice for the parts that come from zeth.  Let me know what you think.

Reply:
The notice looks good, however there is a license mismatch here. LGPL isn't compatible with MIT. That's not a big deal because the LGPL code is ours, but conceptually that's a little weird. How much of the code is a direct copy/paste (or derivative work) to what we have in Zeth?

Reply:
Do we need to keep this code both in Zeth and libff? If not, we can move it to libff and deleted in from the zeth code base, effectively solving the license inconsistency by just licensing these files with MIT.

Reply:
I definitely want to *move* the code.  Obviously, that can't be done until libsnark and zeth are updated to use this version of libff, but I certainly prefer not to maintain 2 versions.

Reply:
> I definitely want to move the code
 
Perfect, then. If we move the code here, I think we don't need to mention that it originally comes from Zeth (it will becom part of libff and the Zeth code will use this updated version of libff). We just relicense the code excerpts moved here. As long as the code moved here - an MIT repo - has been authored **only** by you and/or I, we're good to go -> no need to chase anyone else for their consent to change the license on these lines of code.

Reply:
I've removed the comments about this originating in zeth.
It's all serialization stuff I did recently, so should be no problem from a license point of view.

Reply:
Perfect, thanks